src/github/copilotRemoteAgent.ts

```diff
@@ -751,12 +751,42 @@ export class CopilotRemoteAgentManager extends Disposable {
 		return fullText;
 	}
 
-	public async provideNewChatSessionItem(options: { prompt?: string; history: ReadonlyArray<vscode.ChatRequestTurn | vscode.ChatResponseTurn>; metadata?: any; }, token: vscode.CancellationToken): Promise<ChatSessionWithPR | ChatSessionFromSummarizedChat> {
-		const { prompt, history } = options;
-		if (!prompt) {
+	extractFileReferences(references: readonly ChatPromptReference[] | undefined): string | undefined {
+		if (!references || references.length === 0) {
+			return;
+		}
+		// 'file:///Users/jospicer/dev/joshbot/.github/workflows/build-vsix.yml'  -> '.github/workflows/build-vsix.yml'
+		const parts: string[] = [];
+		for (const ref of references) {
+			if (ref.value instanceof vscode.Uri && ref.value.scheme === 'file') { // TODO: Add support for more kinds of references
+				const workspaceFolder = vscode.workspace.getWorkspaceFolder(ref.value);
+				if (workspaceFolder) {
+					const relativePath = pathLib.relative(workspaceFolder.uri.fsPath, ref.value.fsPath);
+					parts.push(` - ${relativePath}`);
+				}
+			}
+		}
+
+		if (!parts.length) {
+			return;
+		}
+
+		parts.unshift('The user has attached the following files as relevant context:');
+		return parts.join('\n');
+	}
+
+	cleanPrompt(prompt: string): string {
+		// Remove #file:xxxx from the prompt
+		return prompt.replace(/#file:\S+/g, '').trim();
```

Comment:
We want to keep the file name (or relative file path) in the prompt, for example, user might write `add tests to #file:x.ts`, if we replace it completely and include multiple files as relevant files, the prompt itself might be ambiguous as it reads as `add tests to`.